src/types.ts

```diff
@@ -0,0 +1,133 @@
+import type { WebTransport } from './dom'
+
+/**
+ * Native Http3WTSession counterpart
+ */
+ export interface NativeHttp3WTSession {
```

Comment:
Question: this is an internal interface and should not be exported outside of the package, could this be prevented, the same goes for other interfaces that are not part of the spec.

Reply:
Same goes with internal types

Reply:
When you have a JSDoc at the top of the file like:

```js
/**
 * @typedef {import('./dom'). NativeHttp3WTSession} NativeHttp3WTSession
 */
```

it will also export the type from the current file.

With JSDoc types to prevent this we'd need to split the class that references the internal type from the `@typedef` into it's own file, and import that class from the module's entry point.

With pure typescript you can just not export the interface which keeps it internal.

Reply:
Ok, then splitting the file would be a good idea, so that no one tries to hack into it.  Since these interfaces are not guaranteed to stay the same between patch releases. Or do you think it will not cause harm?

Reply:
I would also like the files to be split - I didn't do it as part of this PR as it would make it a lot harder to review..

Reply:
ok then we can make it in a follow up PR?

Reply:
Yeah, I think that's best, the changes here are disruptive enough!

Reply:
ok